tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -626,12 +674,21 @@ def create_manifest_and_resource(tests, id)
     tests[id][:resource] = manifest_props unless tests[id][:resource]
   end
 
-  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
-  \nnode default {
-  #{dependency_manifest(tests, id)}
-  #{tests[:resource_name]} { '#{tests[id][:title_pattern]}':
-    #{state}\n#{manifest}
-  }\n}\nEOF"
+  tests[id][:manifest] = if tests[:agent]
+
+                           "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+                         \nnode default {
+                         #{dependency_manifest(tests, id)}
+                         #{tests[:resource_name]} { '#{tests[id][:title_pattern]}':
+                           #{state}\n#{manifest}
+                         }\n}\nEOF"
+                         else
+                           "cat <<EOF > apply.pp
```

Comment:
maybe define `AGENTLESS_MANIFESTPATH`?

Reply:
maybe replace the entire section with proper ruby
```
File.open('apply.pp', 'w') do |f|
  f.write(...)
end
```

Reply:
Also, http://ruby-doc.org/stdlib-2.1.9/libdoc/tempfile/rdoc/Tempfile.html

Reply:
Using tempfile, which inherently uses file. Not sure why we need AGENTLESS_MANIFESTPATH - the agentful path was because we wanted to define the site.pp on master. I have created a prefix constant for our tempfiles here though.